pkg/apis/trust/v1alpha1/types_bundle.go

```diff
@@ -76,6 +76,14 @@ type BundleSource struct {
 	// InLine is a simple string to append as the source data.
 	// +optional
 	InLine *string `json:"inLine,omitempty"`
+
+	// DefaultPackage, when true, requests the default bundle package to be used as a source.
+	// The default bundle package is available if trust-manager was installed via Helm
+	// or was otherwise set up to include a package-injecting init container.
+	// If the default package was not configured at start-up, a request to use the default
+	// package will fail.
+	// +optional
+	DefaultPackage bool `json:"defaultPackage,omitempty"`
```

Comment:
> What do you think the evolution of the API will be to allow a cluster operator to do that?

I think that would be an entirely different (new) field, added later. At the start of this work I was using something like `FilesystemPackage string` which allowed a user to point at a named package on the filesystem... but it's incredibly awkward to use that because you have to inspect the deployment of trust-manager to work out what container might've written a package, and maybe check logs to see what was written and what it's called. Much worse when it's something that's configured by default.

I think I'm moving away from arbitrary filesystem packages being something that's generally desirable, at least until I see a valid use case where it's an improvement. We're really only using a container to supply a default package because it's easy to change the package while running the same version of trust-manager.

For other non-default "package" types, I can't currently see why I'd want to run a container as a sidecar to trust-manager. I think I'd want a separate pod which writes/maintains ConfigMaps that trust-manager can consume. We can't really do that for default packages but I think that would be preferable for other types. Just IMO.

Reply:
> Fairly sure this need to be a pointer as it is going to default to false and not be omitted in marshelling.

Bool pointers make my heart hurt but I see the reasoning, will change :+1: